clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/model/AppEngineServerGroup.groovy

```diff
@@ -64,8 +71,36 @@ class AppEngineServerGroup implements ServerGroup, Serializable {
 
   @Override
   ServerGroup.Capacity getCapacity() {
-    def instanceCount = instances?.size()
-    new ServerGroup.Capacity(min: instanceCount, max: instanceCount, desired: instanceCount)
+    Integer instanceCount = instances?.size() ?: 0
+
+    switch (scalingPolicy?.type) {
+      case ScalingPolicyType.AUTOMATIC:
+        /*
+        * For the flexible environment, a version using automatic scaling can be stopped.
+        * A stopped version scales down to zero instances and ignores its scaling policy.
+        * */
+        def min = Math.min(instanceCount, scalingPolicy.minTotalInstances)
+        def desired = servingStatus == ServingStatus.SERVING ? scalingPolicy.maxTotalInstances : 0
+        return new ServerGroup.Capacity(min: min ?: 0,
```

Comment:
What's the point of the `?:` here? It seems like `min` can't be null since `instanceCount` will never be `null`.